### ISSUE TITLE
FR: Ability to prevent FanControl window from opening window after waking PC

### ISSUE THREAD

**Issue by hapklaar** [Rem0o/FanControl.Releases#2265]
Every time I wake my PC from sleep, the FanControl window opens and doesn't minimize by itself. Start minimized is enabled.

Can this be prevented somehow, so the window remains minimized to tray?

**Comment by Rem0o** [comment]
Do you get any popup error when the window is opened up automatically?

**Comment by hapklaar** [comment]
Currently there are no errors when resuming from sleep.

Sent from Nine

________________________________
Van: "Rem0o ***@***.***)" ***@***.***> 
verzonden: woensdag 24 januari 2024 02:49 
Aan: Rem0o/FanControl.Releases 
Cc: hapklaar; Author
Onderwerp: Re: [Rem0o/FanControl.Releases] FR: Ability to prevent FanControl window from opening window after waking PC (Issue #2265)

Do you get any popup error when the window is opened up automatically?

**Comment by hapklaar** [comment]
Strangely enough when I just woke the PC, the window did not pop up... Should it only happen when there are errors?

**Comment by Rem0o** [comment]
> Should it only happen when there are errors?

Yes, well, if some part of the program asked to show a dialog, which upon wake from sleep to happen automatically, is an error in the refresh process .

**Comment by hapklaar** [comment]
Today it opened again after resuming from sleep. Nothing in the error log. Window appears like this:
![image](https://github.com/Rem0o/FanControl.Releases/assets/7703320/4cdcaec3-a1c1-4069-9f1f-3697be946806)

The bottom notification `Configuration loaded: userConfig.json` kept dropping down and reappearing. Maybe that is relevant.

**Comment by Rem0o** [comment]
@hapklaar that means the app refreshed for every time that snackbar popped up, which means upon waking from sleep the app can't get hold of all the sensors it had before going to sleep.

**Comment by hapklaar** [comment]
Ah I see. The level of sleep might play a part here.

Would it be possible to do this in the background and not show the screen?

**Comment by Rem0o** [comment]
The intention is to only show the window if it couldn't restore properly, hence functionality was not restored, needing attention.

In your case there seem to be a bug.   I will give a look.

**Comment by Rem0o** [comment]
Any log.txt?

I made a debug version of the exe with logs of the whole wake from sleep sequence, if you could run that, reproduce the wake from sleep problem, then send back the logs.

[Wake from sleep logs.zip](https://github.com/Rem0o/FanControl.Releases/files/14066872/Wake.from.sleep.logs.zip)

**Comment by hapklaar** [comment]
sure, will do. Will it write to log.txt in the exe's directory?

**Comment by Rem0o** [comment]
yes

**Comment by hapklaar** [comment]
Today when I woke the pc from sleep, no FanControl window was opened, but I noticed after a while my radiator fans were spinning louder than they should. I opened FanControl from systray and was greeted with a message that not all sensors were detected. After clicking refresh, this was fixed.

So not the original issue I was facing, but maybe it is still useful.

I enclosed the log
[log.txt](https://github.com/Rem0o/FanControl.Releases/files/14072923/log.txt)

**Comment by Rem0o** [comment]
This tells me it takes more than 30 seconds for your system to be able to get back the sensors after sleep.   When waking from sleep, fancontrol will refresh every 5 seconds for 30 seconds until all sensors that were previously there before sleep are restored.   Maybe in your case that's just not enough, which is surprising.

**Comment by hapklaar** [comment]
> I made a debug version of the exe with logs of the whole wake from sleep sequence, if you could run that, reproduce the wake from sleep problem, then send back the logs.

Just to let you know. I haven't had the issue since I installed your debug version. Could the debugging itself have influence?

**Comment by AzuMiWanHua** [comment]
I have the same problem and occasionally need to manually refresh the sensor, is there a solution now?
